### DISCUSSION TITLE
use of decontXcounts in subseq

### DISCUSSION THREAD

**Discussion by Sa753** [campbio/celda#345] [Q&A]
Dear team,

Thank you for the useful tool. May I ask if I provided a matrix of raw counts to the decontX function and I got the output as a list, what should I do next to use it in seurat object for subsequent analysis?.

For example, should I add it as an assay in seurat and use the counts to perform the clustering and differential expression?

Thanks

**Comment by joshua-d-campbell** [comment]
Hi @Sa753, thanks for your question. Yes, you can take the "decontXcounts" matrix (which is returned as an item in the list) and add it to a Seurat object like you would the original raw counts matrix. The values in the decontXcounts matrix are no longer integers and some people like to round them to make them integers again. You should also assess the levels of contamination. See this tutorial for more information on plots you can make to help with this:

https://celda.camplab.net/v1.9.2/index.html

**Comment by Sa753** [comment]
Thank you so much for prompt reply. So should I use the following code for example

sce <- as.SingleCellExperiment(seuratobject)
sce <- decontx(sce)
seuratobject@assays$RNA@counts <-  sce@assays@data@listData$decontXcounts

then I normalise the decontXcounts as they are now instead of my raw counts 

The reason I am asking about the above code is that,  when I check the plots using the sce object and the plotting functions in celda package, I find that it works very well but when I add the matrix to Seurat object and plot using Vlnplot function in seurat package , I don't see the same results so I thought I maybe putting back the decontXcounts incorrectly in my seuratobejct.
Thanks so much

**Comment by joshua-d-campbell** [comment]
If your data is already in a Seurat object, that would probably work. Something like this might be more straightforward:

```
mat <- Read10X("path/to/matrix")
res <- decontX(mat)
seurat <- CreateSeuratObject(counts = res$decontXcounts, min.cells = 3, min.features = 200)
```

where 'mat' is the sparse counts matrix that can be read in using the Read10X function from Seurat. Then you can move forward with your analysis from there.

**Comment by Sa753** [comment]
Hi Josha,

Thank you so much for that. 

Do you have any suggestions if I have an already integrated seurat object?. I saw this in issue number https://github.com/campbio/celda/issues/334  who used it on the integrated object successfully. 

Thanks so much
